packages/vitest/src/node/core.ts

```diff
@@ -1086,21 +1088,58 @@ export class Vitest {
   }
 
   public async globTestSpecs(filters: string[] = []) {
+    const dir = this.config.dir || this.config.root
+    const parsedFilters = filters.map(f => parseFilter(f))
+    const testLocations = groupFilters(parsedFilters.map(
+      f => ({ ...f, filename: resolve(dir, f.filename) }),
```

Comment:
I wonder if it makes sense to validate that the path exists. Filters can be anything, you can provide a part of the file

I think we should throw an error if:
- There is a location in filters
- File is resolved with `process.cwd(), file`, not `root` (because filters are provides via a CLI normally)
- File doesn't exist, or it's not a file at all (a directory for example)

Reply:
Interesting.

What came to mind is to limit matching to full filename, instead of part of it, in order not to match multiple files at the same time.  Does that sound right to you?

Note that I'm passing the filters as is (before resolving) to `project.globTestFiles`, and then I narrow the matching for filters with location specified.  Later at line 1143, I'm skipping over reporting errors for filters that don't have location specified.

---

For now I'm just reporting error and continuing.  Now that I'm thinking about it, this has weird behavior for example `pnpm exec vitest basic.test.ts:3` would report an error about "not finding the file", but will run the tests anyways.

It seems I might be able to fix this by only resolving filters with test location and then passing on to `project.globTestFiles`, does that sound right to you?

Good point on resolving with `process.cwd()`.

<!-- Note that later at line 1143, I'm skipping over reporting errors for filters that don't have location specified; and I'm passing the filters as is (before resolving) to `globTestFiles`. -->

Reply:
I don't know if I follow what you want to do instead of what I proposed